src/Features/Core/Portable/TodoComments/AbstractTodoCommentService.cs

```diff
@@ -3,16 +3,20 @@
 using System;
 using System.Collections.Generic;
 using System.Collections.Immutable;
+using System.Runtime.CompilerServices;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.CodeAnalysis.Remote;
 using Microsoft.CodeAnalysis.Shared.Extensions;
 using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.TodoComments
 {
     internal abstract class AbstractTodoCommentService : ITodoCommentService
     {
+        private readonly ConditionalWeakTable<RemoteHostClient, KeepAliveSession> _sessions = new ConditionalWeakTable<RemoteHostClient, KeepAliveSession>();
```

Comment:
I don't see how your explanation relates to this code.  Why can't we just have a pool of clients/sessions in the OOP layer that are then handed out to features when they need it?